routes/api/bookin.js

```diff
@@ -467,14 +467,20 @@ router.post('/changeReservation',(req,res)=>{
 router.post('/cancel', (req,res)=>{
     Booking.findById(req.body.bookingID)
     .then(booking => {
-        if (booking.check_in_date - new Date() > 172800000){
-            if (booking.status == 0){
-                booking.status = 3;
-                if (booking.rewardPointsUsed !== 0){
-                    User.findOneAndUpdate(
-                        {'customerID': booking.customerID},
-                        {$inc: { "rewardPoints" : booking.rewardPointsUsed }
-                    })
+            if (booking.status === 0){
+                booking.status = 0;
+                if (booking.check_in_date - new Date > 172800000){
```

Comment:
Should also check if user had make a change to their reservation, if they have, there will be anew check in date